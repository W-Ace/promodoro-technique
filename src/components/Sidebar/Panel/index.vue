<template>
  <div class="panel">
    <div class="panel-header">
      <slot name="header" />
    </div>
    <div class="panel-body">
      <slot name="body" />
    </div>
  </div>
</template>
<style lang="scss" scoped>
.panel {
  width: 518px;
  padding: 30px 33px;
}

.panel-header {
  @include lato-black;

  color: #fcfcfc;
  font-size: 20px;
  letter-spacing: 2px;
}

.panel-body {
  position: relative;
  margin-top: 20px;
  padding-top: 20px;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    display: block;
    width: 100%;
    height: 1px;
    background: #6a6969;
    content: '';
    transform: scaleY(0.5);
  }
}
</style>
