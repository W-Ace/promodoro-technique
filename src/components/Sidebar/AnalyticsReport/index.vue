<template>
  <panel>
    <template #header>
      ANALYTICS REPORT
    </template>
    <template #body>
      1
    </template>
  </panel>
</template>
<script>
import Panel from '@/components/Sidebar/Panel';

export default {
  components: {
    Panel,
  },
};
</script>
