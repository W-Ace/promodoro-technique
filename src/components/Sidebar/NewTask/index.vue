<template>
  <panel>
    <template #header>
      ADD NEW TASK
    </template>
    <template #body>
      1
    </template>
  </panel>
</template>
<script>
import Panel from '@/components/Sidebar/Panel';

export default {
  components: {
    Panel,
  },
};
</script>
