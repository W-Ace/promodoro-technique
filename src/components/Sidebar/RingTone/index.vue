<template>
  <panel>
    <template #header>
      RING TONE
    </template>
    <template #body>
      1
    </template>
  </panel>
</template>
<script>
import Panel from '@/components/Sidebar/Panel';

export default {
  components: {
    Panel,
  },
};
</script>
