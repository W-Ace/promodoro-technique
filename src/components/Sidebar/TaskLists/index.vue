<template>
  <panel>
    <template #header>
      TASK LISTS
    </template>
    <template #body>
      1
    </template>
  </panel>
</template>
<script>
import Panel from '@/components/Sidebar/Panel';

export default {
  components: {
    Panel,
  },
};
</script>
