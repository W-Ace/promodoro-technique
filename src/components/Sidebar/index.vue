<template>
  <aside class="sidebar">
    <ul class="menu">
      <li
        class="menu-item"
        id="new-task"
        :class="{'active': isCurrentMenu('NewTask')}"
        @click="changeMenu('NewTask')"
      >
        <img
          class="img"
          :src="require(`@/assets/img/components/sidebar/add-${getColorOfMenuIcon('NewTask')}.svg`)"
          alt="new-task"
        >
      </li>
      <li
        class="menu-item"
        id="task-lists"
        :class="{'active': isCurrentMenu('TaskLists')}"
        @click="changeMenu('TaskLists')"
      >
        <img
          class="img"
          :src="require(`@/assets/img/components/sidebar/list-${getColorOfMenuIcon('TaskLists')}.svg`)"
          alt="task-lists"
        >
      </li>
      <li
        class="menu-item"
        id="analytics-report"
        :class="{'active': isCurrentMenu('AnalyticsReport')}"
        @click="changeMenu('AnalyticsReport')"
      >
        <img
          class="img"
          :src="require(`@/assets/img/components/sidebar/analytics-${getColorOfMenuIcon('AnalyticsReport')}.svg`)"
          alt="analytics-report"
        >
      </li>
      <li
        class="menu-item"
        id="ring-tone"
        :class="{'active': isCurrentMenu('RingTone')}"
        @click="changeMenu('RingTone')"
      >
        <img
          class="img"
          :src="require(`@/assets/img/components/sidebar/ringtone-${getColorOfMenuIcon('RingTone')}.svg`)"
          alt="ring-tone"
        >
      </li>
    </ul>
    <component :is="menuType" />
  </aside>
</template>
<script>
import * as R from 'ramda';
import { ref } from 'vue';
import NewTask from '@/components/Sidebar/NewTask';
import TaskLists from '@/components/Sidebar/TaskLists';
import AnalyticsReport from '@/components/Sidebar/AnalyticsReport';
import RingTone from '@/components/Sidebar/RingTone';

export default {
  components: {
    NewTask,
    TaskLists,
    AnalyticsReport,
    RingTone,
  },
  setup() {
    const defaultMenu = 'NewTask';
    const menuType = ref(defaultMenu);

    const changeMenu = ((menu) => (type) => {
      menu.value = type;
      return type;
    })(menuType);

    const isCurrentMenu = (type) => R.equals(menuType.value, type);

    const getMenuImgColor = (bool) => (bool ? 'red' : 'white');

    const getColorOfMenuIcon = R.pipe(
      isCurrentMenu,
      getMenuImgColor,
    );
    return {
      menuType,
      changeMenu,
      isCurrentMenu,
      getColorOfMenuIcon,
    };
  },
};
</script>
<style lang="scss" scoped>
.sidebar {
  display: flex;
  background: #333;
}

.menu {
  width: 82px;
}

.menu-item {
  cursor: pointer;

  .img {
    width: 100%;
    height: auto;
  }
}
</style>
