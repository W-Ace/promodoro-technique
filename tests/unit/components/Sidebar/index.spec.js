import { shallowMount } from '@vue/test-utils';
import Sidebar from '@/components/Sidebar';

describe('menu', () => {
  it('Click newTask of menuItem will add class of active on it', async () => {
    const wrapper = shallowMount(Sidebar);
    const menuItem = wrapper.find('#new-task');
    await menuItem.trigger('click');
    expect(menuItem.classes()).toContain('active');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Click taskLists of menuItem will add class of active on it', async () => {
    const wrapper = shallowMount(Sidebar);
    const menuItem = wrapper.find('#task-lists');
    await menuItem.trigger('click');
    expect(menuItem.classes()).toContain('active');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Click analyticsReport of menuItem will add class of active on it', async () => {
    const wrapper = shallowMount(Sidebar);
    const menuItem = wrapper.find('#analytics-report');
    await menuItem.trigger('click');
    expect(menuItem.classes()).toContain('active');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Click ringTone of menuItem will add class of active on it', async () => {
    const wrapper = shallowMount(Sidebar);
    const menuItem = wrapper.find('#ring-tone');
    await menuItem.trigger('click');
    expect(menuItem.classes()).toContain('active');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
