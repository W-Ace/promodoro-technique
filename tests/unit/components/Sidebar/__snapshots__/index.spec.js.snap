// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu Click analyticsReport of menuItem will add class of active on it 1`] = `
<aside class="sidebar">
  <ul class="menu">
    <li class="menu-item" id="new-task"><img class="img" src="" alt="new-task"></li>
    <li class="menu-item" id="task-lists"><img class="img" src="" alt="task-lists"></li>
    <li class="menu-item active" id="analytics-report"><img class="img" src="" alt="analytics-report"></li>
    <li class="menu-item" id="ring-tone"><img class="img" src="" alt="ring-tone"></li>
  </ul>
  <analytics-report-stub></analytics-report-stub>
</aside>
`;

exports[`menu Click newTask of menuItem will add class of active on it 1`] = `
<aside class="sidebar">
  <ul class="menu">
    <li class="menu-item active" id="new-task"><img class="img" src="" alt="new-task"></li>
    <li class="menu-item" id="task-lists"><img class="img" src="" alt="task-lists"></li>
    <li class="menu-item" id="analytics-report"><img class="img" src="" alt="analytics-report"></li>
    <li class="menu-item" id="ring-tone"><img class="img" src="" alt="ring-tone"></li>
  </ul>
  <new-task-stub></new-task-stub>
</aside>
`;

exports[`menu Click ringTone of menuItem will add class of active on it 1`] = `
<aside class="sidebar">
  <ul class="menu">
    <li class="menu-item" id="new-task"><img class="img" src="" alt="new-task"></li>
    <li class="menu-item" id="task-lists"><img class="img" src="" alt="task-lists"></li>
    <li class="menu-item" id="analytics-report"><img class="img" src="" alt="analytics-report"></li>
    <li class="menu-item active" id="ring-tone"><img class="img" src="" alt="ring-tone"></li>
  </ul>
  <ring-tone-stub></ring-tone-stub>
</aside>
`;

exports[`menu Click taskLists of menuItem will add class of active on it 1`] = `
<aside class="sidebar">
  <ul class="menu">
    <li class="menu-item" id="new-task"><img class="img" src="" alt="new-task"></li>
    <li class="menu-item active" id="task-lists"><img class="img" src="" alt="task-lists"></li>
    <li class="menu-item" id="analytics-report"><img class="img" src="" alt="analytics-report"></li>
    <li class="menu-item" id="ring-tone"><img class="img" src="" alt="ring-tone"></li>
  </ul>
  <task-lists-stub></task-lists-stub>
</aside>
`;
